includes/reader-activation/class-reader-activation.php

```diff
@@ -405,6 +405,68 @@ public static function is_esp_configured() {
 	 * @return array
 	 */
 	public static function get_registration_newsletter_lists() {
+		if ( ! class_exists( '\Newspack_Newsletters_Subscription' ) ) {
+			return [];
+		}
+
+		$registration_lists = self::get_available_newsletter_lists();
+
+		/**
+		 * Filters the newsletters lists that should be rendered during registration.
+		 *
+		 * @param array $registration_lists Array of newsletter lists.
+		 */
+		return apply_filters( 'newspack_registration_newsletters_lists', $registration_lists );
+	}
+
+	/**
+	 * Get the newsletter lists that should be rendered after checkout.
+	 *
+	 * @param string $email Email address.
+	 *
+	 * @return array
+	 */
+	public static function get_post_checkout_newsletter_lists( $email ) {
+		$available_lists    = self::get_available_newsletter_lists();
+		$registration_lists = [];
+
+		if ( empty( $available_lists ) || ! method_exists( '\Newspack_Newsletters_Subscription', 'get_contact_lists' ) ) {
+			return [];
+		}
+
+		$current_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $email );
+		if ( \is_wp_error( $current_lists ) || ! is_array( $current_lists ) ) {
+			return [];
+		}
+
+		foreach ( $available_lists as $list_id => $list ) {
+			// Skip any lists the reader is already signed up for.
```

Comment:
At the last testing step, I _do_ see the list I've signed up for. But I assume this is because I'm using MC with double opt-in.

Reply:
Hmm. To clarify, did the account you use for testing successfully complete the double opt-in process and get added to the audience in MC? I believe the `Newspack_Newsletters_Subscription::get_contact_lists` should return an array of list IDs the provided email is added to regardless of double or single opt in.

Reply:
> did the account you use for testing successfully complete the double opt-in process

No. 

> should return an array of list IDs the provided email is added to regardless of double or single opt in.

In my testing at least, the contact is not added in MC until they confirm the opt-in via email. That's the purpose of the double opt-in: allow only the email owner to sign up. 

Anyway, disregard my comment, it only muddles things up. This PR is not about the `Newspack_Newsletters_Subscription::get_contact_lists` method.